src/engine/runtime.js

```diff
@@ -1232,63 +1256,102 @@ class Runtime extends EventEmitter {
             argTypeInfo = context.categoryInfo.customFieldTypes[argInfo.type].argumentTypeInfo;
         }
 
-        const defaultValue =
-            typeof argInfo.defaultValue === 'undefined' ? '' :
-                xmlEscape(maybeFormatMessage(argInfo.defaultValue, this.makeMessageContextForTarget()).toString());
+        // Start to construct the scratch-blocks style JSON defining how the block should be
+        // laid out
+        let argJSON;
 
-        if (argTypeInfo.check) {
-            argJSON.check = argTypeInfo.check;
+        // Most field types are inputs (slots on the block that can have other blocks plugged into them)
+        // check if this is not one of those cases. E.g. an inline image on a block.
+        switch (argTypeInfo.fieldType) {
+        case 'field_image': {
+            if (!argInfo.dataURI) {
+                log.warn('Missing data URI in extension block with argument type IMAGE');
+            }
+            argJSON = {
+                type: 'field_image',
+                src: argInfo.dataURI || '',
+                // TODO these probably shouldn't be hardcoded...?
+                width: 24,
+                height: 24,
+                // Whether or not the inline image should be flipped horizontally
+                // in RTL languages. Defaults to false, indicating that the
+                // image will not be flipped.
+                flip_rtl: argInfo.flipRTL || false
+            };
+            break;
         }
+        default: {
```

Comment:
I'm a little uncomfortable with how much code is in this `default: { ... }` here. I see that `'field_image'` needs very different treatment from the rest but... hmm. Does the `default` section use too many of the surrounding variables for it to be pulled out into its own function reasonably? If this really is the best organization then I can probably cope ;)